### PR TITLE
Speed up MultiWriterIdGenerator when lots of IDs are in flight.

### PR DESCRIPTION
**Pull request by erikjohnston**
If we have lots of `_unfinished_ids` in flight (which happens when persisting large numbers of events at once, e.g. during room joins) then its more efficient to keep the set sorted than repeatedly calling `min(_unfinished_ids)`

### CODE REVIEW

File: synapse/storage/util/id_generators.py [hunk 1]
```diff
@@ -473,7 +474,7 @@ def _mark_id_as_finished(self, next_id: int):
 
                 finished = set()
 
-                min_unfinshed = min(self._unfinished_ids)
+                min_unfinshed = self._unfinished_ids[0]
```
Comment (clokep) [inline synapse/storage/util/id_generators.py:477]: If we only care about the minimum one, would it make more sense to track that separately? (Does `SortedSet` have any overhead compared to `set`?)

Comment (erikjohnston) [reply synapse/storage/util/id_generators.py:477]: Potentially, but in practice this change stopped the ID generators from appearing in the profiling, so :shrug:

File: stubs/sortedcontainers/sortedset.pyi [hunk 2]
```diff
@@ -0,0 +1,118 @@
+# stub for SortedDict. This is a lightly edited copy of
```
Comment (clokep) [inline stubs/sortedcontainers/sortedset.pyi:1]:
```suggestion
# stub for SortedSet. This is a lightly edited copy of
```

What was edited? 😄 I didn't look super closely at these, but they look reasonable overall.

**Review by clokep — APPROVED** [review]

File: stubs/sortedcontainers/sortedset.pyi (hunk 2 above)
Comment (erikjohnston) [reply stubs/sortedcontainers/sortedset.pyi:1]: I just isort/blacked it and copied that stub from the other files :)

**Pull request merged**